### PR TITLE
docs: (IAC-352) Subnet requirements need more clarity in doc

### PR DESCRIPTION
**Pull request by dhoucgitter**
# Changes
Update documentation to clarify what the AWS EKS subnet requirements are with and without BYO subnets.
Intentionally based off staging branch to pick up the latest doc changes there.
# Tests
Ran the following tests to verify that added documentation was accurate.
| Scenario | subnet_ids map | external db configured | rds multi_az value | tf apply Notes |
|---|---|---|---|---|
| 1 | 1 private, 2 control plane (unique AZs) | yes | false | fails with Error <sup>1</sup> |
| 2 | 1 private, 2 control plane (unique AZs) | yes | true | fails with Error <sup>1</sup> |
| 3 | 1 private, 2 control plane (unique AZs), 2 database (unique AZs) | yes | true | successful |
| 4 | 1 private, 2 control plane (unique AZs), 2 public (unique AZs) | yes | false | successful |
| 5 | 1 private, 2 control plane (unique AZs) | no | false | successful |

Error <sup>1</sup>

> ╷
> │ Error: creating RDS DB Subnet Group (dh-619-default-pgsql-20231107223512640800000007): DBSubnetGroupDoesNotCoverEnoughAZs: The DB subnet group doesn't meet Availability Zone (AZ) coverage requirement. Current AZ coverage: us-west-1a. Add subnets to cover at least 2 AZs.
> │       status code: 400, request id: e20f9fdf-ebde-4864-b457-d7a4ea6cc817
> │
> │   with module.postgresql["default"].module.db_subnet_group.aws_db_subnet_group.this[0],
> │   on .terraform/modules/postgresql/modules/db_subnet_group/main.tf line 8, in resource "aws_db_subnet_group" "this":
> │    8: resource "aws_db_subnet_group" "this" {
> │
> ╵

Reason for the failure to create RDS DB Subnet Group:

Even with the RDS module input parameter set to multi_az=false,  a DB subnet group has to have multiple subnets. It's a requirement of AWS RDS that you can't bypass.

### CODE REVIEW

**Review by thpang — APPROVED** [review]

**Review by jarpat — APPROVED** [review]

File: docs/CONFIG-VARS.md [hunk 1]
```diff
@@ -111,6 +111,12 @@ You can also use `default_private_access_cidrs` to apply the same CIDR range to
 | subnets | Subnets to be created and their settings | map | See below for default values | This variable is ignored when `subnet_ids` is set (AKA bring your own subnets). All defined subnets must exist within the VPC address space. |
 | subnet_azs | Configure specific AZs you want the subnets to be created in. The values must be distinct | optional map | {} see below for an example | If you wish for the codebase to lookup a list of AZs for you: either don't define subnet_azs to lookup AZs for all subnets, or omit the key for a specific subnet in the map to perform a lookup for it. This variable is ignored when `subnet_ids` is set (AKA bring your own subnets).|
 
+### Subnet requirements
```
Comment (sayeun) [inline docs/CONFIG-VARS.md:114]: This heading does not show up in TOC.

File: docs/CONFIG-VARS.md [hunk 2]
```diff
@@ -111,6 +111,12 @@ You can also use `default_private_access_cidrs` to apply the same CIDR range to
...
+
+For [BYO networking scenarios](../docs/user/BYOnetwork.md#supported-scenarios-and-requirements-for-using-existing-network-resources), existing subnet values are provided using the `subnet_ids` configuration variable. When configuring an EKS cluster without public access, the minimum subnet requirements include a `private` subnet with a recommended CIDR range of /18, and two `control_plane` subnets with recommended CIDR ranges of /28, each in a separate AZ. If an external postgres server is configured, two `database` subnets with recommended CIDR ranges of /25, each in a separate AZ also need to be provided. Amazon RDS relies on having two subnets in separate AZs to provide database redundancy should a failure occur in one AZ. If public access is required for your postgres server or cluster, supply two `public` subnets in different AZs instead of the `database` subnets.
+
 The default values for the subnets variable are as follows:
 
 ```yaml
```
Comment (sayeun) [inline docs/CONFIG-VARS.md:122]: There is a request to remove the mentioning of multi-zonal cluster.

File: docs/CONFIG-VARS.md (hunk 1 above)
Comment (dhoucgitter) [reply docs/CONFIG-VARS.md:115]: added heading to TOC

File: docs/CONFIG-VARS.md (hunk 2 above)
Comment (dhoucgitter) [reply docs/CONFIG-VARS.md:123]: I expect that merging with the updates already made in https://github.com/sassoftware/viya4-iac-aws/pull/241 will take care of that.

**Pull request merged**